### ISSUE TITLE
Training accuracy may suddenly fall to near 0 in the early training stage using pytorch 2.9 and AMP (bfloat16)

### ISSUE THREAD

**Issue by yw155** [pytorch/pytorch#173066]
### 🐛 Describe the bug

I am training a deep convolutional neural network for a custom image classification task using ConvNext-v2 backbone. In the early training stage, for example in warmup or the first 80 epochs, the training accuracy may suddenly fall to near 0 and then recover to the original accuracy in a few epochs. I use AdamW optimizer with AMP (bfloat16), and learning rate scheduling with linear warmup and Cosine annealing, and gradients accumulation to accelerate training on a H20 gpu. The batch size is 512, and the accumulated batch size is 4096. The pytorch and CUDA version are 2.9.0 and 12.8, respectively. I have tried to extend the warmup epochs or decrease the learning rate but failed.

Curiously if I add the wandb watch function in the training loop, this phenomenon will not occur.

However, this phenomenon has never occurred before when running the same code and the same training strategy with batchsize=128 and accumulated  batchsize=1024, and pytorch+CUDA=2.6.0+12.4 on a 3090 GPU.

What are the possible reasons?

### Versions

Pytorch 2.9.0
CUDA 12.8
Python 3.12

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

**Comment by janeyx99** [comment]
This may be a better question for our forum: https://discuss.pytorch.org/
